floor_generator/lib/misc/extension/dart_object_extension.dart

```diff
@@ -23,12 +31,20 @@ extension DartObjectExtension on DartObject {
   /// get the ForeignKeyAction this enum represents,
   /// or the result of `null` if the enum did not contain a valid value
   ForeignKeyAction? toForeignKeyAction() {
-    final enumValueString = toEnumValueString();
-    if (enumValueString == null) {
-      return null;
-    } else {
-      return ForeignKeyAction.values.singleWhereOrNull(
-          (foreignKeyAction) => foreignKeyAction.toString() == enumValueString);
+    final interfaceType = type as InterfaceType;
+    final enumFields = interfaceType.element.fields
+        .where((element) => element.isEnumConstant)
+        .toList();
+
+    // Find the index of the matching enum constant.
+    ForeignKeyAction? enumValue;
+    for (int i = 0; i < enumFields.length; i++) {
+      if (enumFields[i].computeConstantValue() == this) {
+        enumValue = ForeignKeyAction.values[i];
+        break;
+      }
     }
+
+    return enumValue;
```

Comment:
By the way, you can use `getField('_name')?.toStringValue()` to get the Enum name and `getField('index')?.toIntValue()` to find the index.

I think it's the best option to get enum name in this case.

Reply:
Come on... 😁 We use reflection when generating code, so working with private fields is the norm.

Reply:
Hehe true! Cleaned it up just now, should be good to go!

Reply:
i think we can safely merge this pr as the new logic is completely covered. codecov sometimes behaves a bit funky.